src/v/config/property.h

```diff
@@ -185,15 +185,14 @@ class property : public base_property {
         return validate(v);
     }
 
-    void reset() override { _value = default_value(); }
-
-    property<T>& operator()(T v) {
-        _value = std::move(v);
-        return *this;
+    void reset() override {
+        auto v = default_value();
+        update_value(std::move(v));
     }
 
     base_property& operator=(const base_property& pr) override {
-        _value = dynamic_cast<const property<T>&>(pr)._value;
+        auto v = dynamic_cast<const property<T>&>(pr)._value;
+        update_value(std::move(v));
         return *this;
     }
```

Comment:
good catch. presumably this is fine, but it feels really error prone to be able to use operator= to assign a value like this, in which case, would only be changed on the current core and not all cores. i wonder how many users there are of this operator=.

Reply:
It's used in patch_cluster_config (and hopefully nowhere else) to create a copy of the configuration to validate incoming upserts https://github.com/redpanda-data/redpanda/blob/0afd0bd1ab7a012149b460639fe525d82299992f/src/v/redpanda/admin/server.cc#L1702-L1707

given this use, update_value isn't strictly needed here (the bindings are empty in that function)

Reply:
got it, thanks. then it seems ok. the operator= is still weird but we can address it in the future if/when we redesign the config system.